### PR TITLE
Integration test cases for password and username provisioning when JIT

### PR DESCRIPTION
**Pull request by Megala21**
Integration test cases for https://github.com/wso2/product-is/issues/3082
Related PRs - https://github.com/wso2/carbon-identity-framework/pull/1629 and  https://github.com/wso2-extensions/identity-governance/pull/229

**This PR need to be merged after merging https://github.com/wso2/carbon-identity-framework/pull/1629 and extensions/identity-governance/pull/229 and after upgrading carbon-identity-framework version and identity-governance version.**

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/provisioning/JustInTimeProvisioningTestCase.java [hunk 1]
```diff
@@ -72,14 +100,146 @@ public void testAttributeConsumerIndex() throws Exception {
...
+        propertyDTOs[3].setName("password");
+        propertyDTOs[3].setValue(DB_USER_PASSWORD);
+
+        UserStoreDTO userStoreDTO = userStoreConfigAdminServiceClient
+                .createUserStoreDTO(jdbcClass, DOMAIN_ID, propertyDTOs);
+        userStoreConfigAdminServiceClient.addUserStore(userStoreDTO);
+        Thread.sleep(5000);
+        userStoreConfigUtils.waitForUserStoreDeployment(userStoreConfigAdminServiceClient, DOMAIN_ID);
     }
 }
```
Comment (nuwandiw) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/provisioning/JustInTimeProvisioningTestCase.java]: Add new line at the end of the file

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/provisioning/JustInTimeProvisioningTestCase.java [hunk 2]
```diff
@@ -72,14 +100,146 @@ public void testAttributeConsumerIndex() throws Exception {
...
+
+    }
+
+    @Test(priority = 7, groups = "wso2.is", description = "test just in time provisioning with only prompt consent")
+    public void testSAMLToSAMLFederationWithPromptConsent() throws Exception {
+
+        try {
+            updateIdentityProviderJitConfiguration(false, false, true, UserCoreConstants.PRIMARY_DEFAULT_DOMAIN_NAME);
+            super.testSAMLToSAMLFederation();
+            Assert.assertTrue(userStoreClient.isExistingUser(getFederatedTestUser()), "User addion failed with JIT "
```
[the remark this answers is not in the document]
Comment (Megala21) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/provisioning/JustInTimeProvisioningTestCase.java]: Yes. Fixed it. Thanks.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/provisioning/JustInTimeProvisioningTestCase.java [hunk 3]
```diff
@@ -72,14 +100,146 @@ public void testAttributeConsumerIndex() throws Exception {
...
+
+        try {
+            updateIdentityProviderJitConfiguration(true, true, true, UserCoreConstants.PRIMARY_DEFAULT_DOMAIN_NAME);
+            super.testSAMLToSAMLFederation();
+            userMgtClient.updateUserListOfRole(INTERNAL_LOGIN_ROLE, new String[] { Utils.MODIFIED_USER_NAME }, null);
+            checkNewlyCreatedUserLogin(Utils.MODIFIED_USER_NAME);
+        } finally {
+            userStoreClient.deleteUser(Utils.MODIFIED_USER_NAME);
+        }
+
```
Comment (vihanga-liyanage) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/provisioning/JustInTimeProvisioningTestCase.java]: Can we remove this line, please?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/util/Utils.java [hunk 4]
```diff
@@ -144,6 +149,37 @@ public static HttpResponse sendPOSTClaimMessage(HttpResponse response, String co
         return httpClient.execute(post);
     }
 
+    /**
+     * To send the response to jit post authentication handler.
+     * @param response Relevant response.
```
Comment (vihanga-liyanage) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/util/Utils.java]: Format the doc comment. There should be a new line after the header. Also, it'd be great if we can have all comments as single complete sentences.

File: modules/integration/tests-integration/tests-backend/src/test/resources/testng.xml [hunk 5]
```diff
@@ -37,10 +37,10 @@
         </classes>
     </test>
     <!--<test name="is-tests-notification-mgt" preserve-order="true" parallel="false" group-by-instances="true">-->
-    <!--<classes>-->
+        <!--<classes>-->
```
Comment (vihanga-liyanage) [inline modules/integration/tests-integration/tests-backend/src/test/resources/testng.xml]: Are these changes required here?
